### ISSUE TITLE
[Question] Is it possible to accept exception that are not utf-8?

### ISSUE THREAD

**Issue by laojianzi** [ClickHouse/ch-go#226]
https://github.com/ClickHouse/ch-go/blob/e0933a2bd42320a2fc46b8b9f761059cb53e16f0/proto/reader.go#L144-L154

If the user enters some incorrect query statement

e.g: 
```sql
SELECT 世界;
```

I can never get a specific exception message in this case

---

Worse still, the `reader` will be retained and interfere with the next query :-(

**Comment by ernado** [comment]
Can you add an reproducer how to trigger this exception?

This exception was added to catch bugs in binary protocol deserialization, e.g. we are trying to read as string something that appears as not string.

**Comment by laojianzi** [comment]
> Can you add an reproducer how to trigger this exception?
> 
> This exception was added to catch bugs in binary protocol deserialization, e.g. we are trying to read as string something that appears as not string.

---

I think this exception is raised in all syntax error statements with non-utf-8.

demo:
```go
package ch_go_test

import (
	"database/sql"
	"testing"
	"time"

	_ "github.com/ClickHouse/clickhouse-go/v2"
)

func TestCH_GO_NonUTF_8(t *testing.T) {
	tests := []struct {
		name    string
		args    string
		wantErr bool
	}{
		{"carrying non-utf-8 syntax errors", "SELECT 错误 AS number", true},
		{"normal sql query", "SELECT 1 AS number", false}, // FIXME: should get number 1
	}

	conn, err := sql.Open("clickhouse", "clickhouse://xxx")
	if err != nil {
		t.Error(err)
		return
	}

	conn.SetMaxIdleConns(5)
	conn.SetMaxOpenConns(10)
	conn.SetConnMaxLifetime(time.Minute * 3)
	if err = conn.Ping(); err != nil {
		t.Error(err)
		return
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			rows, err := conn.Query(tt.args)
			if err != nil {
				t.Log(err)
			} else {
				for rows.Next() {
					var number int64
					if err = rows.Scan(&number); err != nil {
						t.Error(err)
						return
					}

					t.Log("number:", number)
				}
			}

			if err != nil != tt.wantErr {
				t.Errorf("wantErr = %v but want = %v", tt.wantErr, err != nil)
			}
		})
	}
}

```

**Comment by ernado** [comment]
Fixed in `v0.52.1`, thank you!
